src/sparsezoo/objects/directories.py

```diff
@@ -283,14 +284,90 @@ class OnnxGz(Directory):
     available however, when the `path` property is accessed, it will point only
     to the `model.onnx` as this is the expected behavior for loading an onnx model
     with or without external data.
+
+    Class Invariants:
+        - `self.name` attribute of this class will point to the name of the onnx file
+        - `self._path` and `self.path` will point to the path of the extracted
+            onnx model
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def __repr__(self):
+        return f"OnnxGz(name={self.name}, path={self._path})"
+
     @property
     def path(self):
-        _ = super().path  # call self.path to download initial file if not already
+        """
+        Assumptions:
+            - the tarball contains only one onnx model, with the
+                name `model.onnx`
```

Comment:
This is not the case for the codegen models on sparsezoo, for at least one of them the tarball contains "model_nocache.onnx" Could we just search the directory for a .onnx file instead? We would still be assuming there is only one onnx model file in the folder

Reply:
This is something we want to enforce with `model.onnx.tar.gz` (that it always contains a `model.onnx` along with external data files if any) I could see a future where we support other model names, but they would be handled separately and not via `model.onnx.tar.gz`, the codegen 2B model you mentioned has to be updated on sparsezoo

Reply:
the other case where this may change quickly is for any pipeline requiring more than one model (encoder/decoder pipelines). We haven't come up with a definite plan for these on sparsezoo but I think *.onnx would be more generic?

Reply:
Let's leave it until we have a definite plan for these on sparsezoo; don't want to make any decisions here that are bound to change

Reply:
Done!